### DISCUSSION TITLE
Allow printing or exporting of entire document collection

### DISCUSSION THREAD

**Discussion by designatedsuccessor** [outline/outline#7149] [Ideas]
This is somewhat of a bizarre request, but here goes: I work in a corporate environment that has a requirement where some departments must keep physical (paper) copies of their documentation for business-continuity purposes (e.g., if the world erupts in flames and we have no computers, we need physical copies of procedures).  Outline users can currently print documents one-by-one from Outline, but ideally, they'd like to be able to print whole collections at a time.  That way, every 6 months, they can just go to their 5 collections and print the entire collection; this would allow them to click "Print" 5 times rather than 200 times.

Another way would be to have a "Download" option for an entire collection.  That way, the user could export the entire collection to a single PDF and print that PDF.

**Comment by tommoor** [comment]
Are you using self-hosted or cloud? I think the main limiting factor on this _type_ of request is the volume of data that potentially needs to be gathered in order to generate such an artifact. It's similar to a bulk export, which is something that runs as a background job.

**Reply by designatedsuccessor** [reply]
Cloud.  Wouldn't it basically be the same amount of data of users doing the same thing themselves?

**Reply by tommoor** [reply]
Total amount is the same yes, but all at once. 

Not saying it won't be implemented but just that it's something of a technical challenge to do it correctly

**Reply by designatedsuccessor** [reply]
Is the bulk export human-readable?  If so, that might technically meet our needs, anyway.

**Comment by tommoor** [comment]
Definitely, you can export as HTML and open in the web browser

**Reply by designatedsuccessor** [reply]
This dumb request has come up again at my org, unfortunately.

In the context of the cloud version, do you see the problem more as (a) even a single export to PDF would take a lot of memory or (b) the PDF would have to be recreated any time a document was added/changed?

If the latter, could this be somewhat mitigated by either (a) automatically updating the PDF only once per day or (b) forcing the editor user to click a button every time they want the PDF to be updated?

Another thing I'm looking into is seeing if I can find an offline Markdown-to-PDF combiner program.  I could at least then periodically upload the PDF.  Our users don't really need the PDF to be updated to-the-second.

**Comment by designatedsuccessor** [comment]
Okay, dumb update: I'm using Pandoc to do the conversion, and it seems to work to some degree at least.

One of the biggest obstacles I'm running into is the order of the documents.  When the documents are exported to Markdown, they're not in any particular order, whereas the order of documents in Outline was intentionally established and has contextual meaning.

Example: imagine I have these documents in Outline:

My First Doc
My Second Doc
My Third Doc
My Fourth Doc
My Fifth Doc

When exported to *.md files, there is no way to tell which order they were in when in Outline, aside from manually checking.

Could some type of metadata (or something in the filenames?) be added to indicate order?  Example:
1-My First Doc.md
2-My Second Doc.md
3-My Third Doc.md
...

This would also include nested documents, so their associated exported folders would have similar metadata to indicate order.

**Reply by designatedsuccessor** [reply]
Another thing I've noticed: there is seemingly no difference between titles of documents and H1 headers?  Example:

Title:
`# Title Goes Here`

H1:
`# First H1`

Is that expected behavior?  I guess Markdown has no equivalent to a "document title?"

**Reply by tommoor** [reply]
> Is that expected behavior? I guess Markdown has no equivalent to a "document title?"

Exactly, yea. You can safely assume that every export will have a H1 that represents the document title as the first line.

**Reply by joetabers** [reply]
As far as order goes, some markdown systems that allow reordering save a `.order` file that lists all the files in the directory according to the user's specified order.

**Comment by designatedsuccessor** [comment]
I spent a portion of today looking for a way to automate ordering the documents, but I'm not seeing anything in the export that gives clues about document order.

**Reply by tommoor** [reply]
There isn't anything in the export, I think you'd really have to hit the API to get that information – it's stored in the collection itself, so [collections.documents](https://www.getoutline.com/developers#tag/collections/POST/collections.documents) endpoint would be the best.

**Reply by joetabers** [reply]
It might be worth giving the documents an explicit number prefix to set a consistent order.

**Comment by kylebial0** [comment]
Is there any plan for this feature to be implemented into Outline? This is also a requirement for our company because some people insist on having old school hard copies of everything instead of modern, always up to date, drill-down digital documentation.

**Reply by tommoor** [reply]
It would be nice for sure, devil is in the detail – as you might have to potentially download hundreds of documents and all their images to be able to do an accurate print

**Reply by joetabers** [reply]
I don't know if I'd want to throw out the whole book if only a few documents changed. It would be nice if collections.export could filter to documents updated since some specific date.

**Comment by designatedsuccessor** [comment]
I hate to revisit this and be a pest, but is there any way this might be a feature?  I'm getting increased pressure from my users for this.  At this point, I think we'd be okay with it being locked to only admins to reduce the likelihood of abuse (i.e., killing server resources because people hit the export button over and over).

What I'm being asked for now is a monthly PDF of an entire document collection, which would then be exported to a SharePoint site, whose users would do whatever the hell it is that they need to do with the PDF version.

Rather than hitting a button, could the feature instead be something like "Automatically this export document collection to PDF and email the PDF file to these email address(es) on this schedule"?  That way, I would think there would be much less likely abuse and less likely running out of server resources.

**Reply by tommoor** [reply]
> any way this might be a feature?

Yes, it makes total sense – just behind a bunch of other things that are getting prioritized right now

**Reply by designatedsuccessor** [reply]
My 6-month pester...👀  Any chance this could happen in the next few months?
